src/20231229_Multi_TreasuryManagementGSMFundingRWAStrategyPreparationsPart1/AaveV3Polygon_TreasuryManagementGSMFundingRWAStrategyPreparationsPart1_20231229.sol

```diff
@@ -0,0 +1,118 @@
+  function bridge(address token, uint256 amount) external;
+}
+
+/**
+ * @title Treasury Management - GSM Funding & RWA Strategy Preparations (Part 1)
+ * @author efecarranza.eth
+ * - Snapshot: https://snapshot.org/#/aave.eth/proposal/0xb39537e468eef8c212c67a539cdc6d802cd857f186a4f66aefd44faaadd6ba19
+ * - Discussion: https://governance.aave.com/t/arfc-treasury-management-gsm-funding-rwa-strategy-preparations/16128
+ */
+contract AaveV3Polygon_TreasuryManagementGSMFundingRWAStrategyPreparationsPart1_20231229 is
+  IProposalGenericExecutor
+{
+  using SafeERC20 for IERC20;
+
+  uint256 public constant AAVE_V3_DAI_TO_WITHDRAW = 1_000_000 ether;
+
+  IAavePolEthERC20Bridge public constant bridge =
+    IAavePolEthERC20Bridge(MiscPolygon.AAVE_POL_ETH_BRIDGE);
+
+  function execute() external {
+    AaveV3Polygon.COLLECTOR.transfer(
+      AaveV2PolygonAssets.USDC_A_TOKEN,
+      address(this),
+      IERC20(AaveV2PolygonAssets.USDC_A_TOKEN).balanceOf(address(AaveV3Polygon.COLLECTOR)) - 10e6
+    );
+
+    AaveV3Polygon.COLLECTOR.transfer(
+      AaveV2PolygonAssets.USDT_A_TOKEN,
+      address(this),
+      IERC20(AaveV2PolygonAssets.USDT_A_TOKEN).balanceOf(address(AaveV3Polygon.COLLECTOR)) - 10e6
+    );
+
+    AaveV3Polygon.COLLECTOR.transfer(
+      AaveV2PolygonAssets.DAI_A_TOKEN,
+      address(this),
+      IERC20(AaveV2PolygonAssets.DAI_A_TOKEN).balanceOf(address(AaveV3Polygon.COLLECTOR)) - 10e18
+    );
+
+    AaveV3Polygon.COLLECTOR.transfer(
+      AaveV3PolygonAssets.USDC_A_TOKEN,
+      address(this),
+      IERC20(AaveV3PolygonAssets.USDC_A_TOKEN).balanceOf(address(AaveV3Polygon.COLLECTOR)) - 10e6
+    );
+
+    AaveV3Polygon.COLLECTOR.transfer(
+      AaveV3PolygonAssets.USDT_A_TOKEN,
+      address(this),
+      IERC20(AaveV3PolygonAssets.USDT_A_TOKEN).balanceOf(address(AaveV3Polygon.COLLECTOR)) - 10e6
+    );
+
+    AaveV3Polygon.COLLECTOR.transfer(
+      AaveV3PolygonAssets.DAI_A_TOKEN,
+      address(this),
+      AAVE_V3_DAI_TO_WITHDRAW
+    );
+
+    AaveV2Polygon.POOL.withdraw(
+      AaveV3PolygonAssets.USDC_UNDERLYING,
+      type(uint256).max,
+      address(this)
+    );
+
+    AaveV2Polygon.POOL.withdraw(
+      AaveV3PolygonAssets.USDT_UNDERLYING,
+      type(uint256).max,
+      address(this)
+    );
+
+    AaveV2Polygon.POOL.withdraw(
+      AaveV3PolygonAssets.DAI_UNDERLYING,
+      type(uint256).max,
+      address(this)
+    );
+
+    AaveV3Polygon.POOL.withdraw(
+      AaveV3PolygonAssets.USDC_UNDERLYING,
+      type(uint256).max,
+      address(this)
+    );
+
+    AaveV3Polygon.POOL.withdraw(
+      AaveV3PolygonAssets.USDT_UNDERLYING,
+      type(uint256).max,
+      address(this)
+    );
+
+    AaveV3Polygon.POOL.withdraw(
+      AaveV3PolygonAssets.DAI_UNDERLYING,
+      AAVE_V3_DAI_TO_WITHDRAW,
+      address(this)
+    );
+
+    bridge.bridge(
+      AaveV2PolygonAssets.USDC_UNDERLYING,
+      IERC20(AaveV3PolygonAssets.USDC_UNDERLYING).balanceOf(address(AaveV3Polygon.COLLECTOR))
```

Comment:
This is bridging 0 USDC, as the USDC balance of collector will be 0 (Same for DAI and USDT as well).
Instead of querying the balance of collector here, we should be querying the balance of the bridge contract.

Please note `address(this)` on the payload represents the executor lvl1 address as the payload is being called via delegatecall.

Also when we bridge we assume the tokens to be present in the bridge contract, so when withdrawing above we can do as follow:

```
AaveV2Polygon.POOL.withdraw(
      AaveV3PolygonAssets.USDC_UNDERLYING,
      type(uint256).max,
      address(bridge)
    );
```
Instead of withdrawing to `address(this)`